core/src/main/scala/org/bitcoins/core/script/stack/StackInterpreter.scala

```diff
@@ -69,10 +68,10 @@ sealed abstract class StackInterpreter {
     require(program.script.headOption.contains(OP_TOALTSTACK),
             "Top of script stack must be OP_TOALTSTACK")
     if (program.stack.nonEmpty) {
-      ScriptProgram(program,
-                    program.stack.tail,
-                    program.script.tail,
-                    program.stack.head :: program.altStack)
+      program
+        .updateStack(program.stack.tail)
```

Comment:
nit: This could be
```
program.
updateStackAndScript(program.stack.tail, program.script.tail).
updateAltStack(program.stack.head :: program.altStack)
```

Not sure what you guys would consider cleaner